### PR TITLE
Propose trait associated items and autoimport traits on completion

### LINKED ISSUE #7248: Feature request: Automatic import for trait methods
Using IntelliJ Rust, we can type
```rs
v.iter().tu
```
and be prompted with `tuple_combinations` from the itertools crate.
image

With rust-analyzer, we do not get prompted with `tuple_combinations`.
image

### PR DESCRIPTION
**Pull request by SomeoneToIgnore**
trait_imports

Closes #7248

### CODE REVIEW

File: crates/completion/src/completions/flyimport.rs [hunk 1]
```diff
@@ -60,58 +65,108 @@ use crate::{
...
         return None;
     }
-    let potential_import_name = ctx.token.to_string();
-    if potential_import_name.len() < 2 {
-        return None;
-    }
-    let _p = profile::span("import_on_the_fly").detail(|| potential_import_name.to_string());
+    let potential_import_name = {
+        let token_kind = ctx.token.kind();
+        if token_kind == T![.] || token_kind == T![::] {
```
Comment (matklad) [inline crates/completion/src/completions/flyimport.rs:76]:
```suggestion
        if matches!(token_kind, T![.] |T![::]) {
```

**Review by matklad — COMMENTED** [review]
r=me

:tada:

**Comment by SomeoneToIgnore** [comment]
bors r=matklad

Here we go 🎉

**Comment by SomeoneToIgnore** [comment]
Formatting was a bit off.

bors retry

**Pull request merged**